sdk/extensions/Microsoft.Extensions.Azure/src/Internal/ClientFactory.cs

```diff
@@ -119,12 +124,13 @@ internal static TokenCredential CreateCredential(IConfiguration configuration)
             {
                 int idCount = 0;
                 idCount += string.IsNullOrWhiteSpace(clientId) ? 0 : 1;
+                idCount += string.IsNullOrWhiteSpace(managedIdentityClientId) ? 0 : 1;
```

Comment:
Actually I don't think we need this, as the other 3 already support [all the identifiers](https://github.com/Azure/azure-sdk-for-net/pull/50436/files#r2127560519)

Reply:
Yes, good point. We need to support both clientId (for the app) and managedIdentityClientId for the MI in this federated scenario. So we can't require them to be mutually exclusive if the `managedidentityasfederatedidentity` was configured.

Reply:
Talked this over offline, and this is safe because of the check on L123.